util/unifyfs/src/unifyfs-rm.c

```diff
@@ -807,11 +808,27 @@ int unifyfs_start_servers(unifyfs_resource_t* resource,
         return rc;
     }
 
+
+
     if (args->script != NULL) {
-        return script_launch(resource, args);
+        return_val=script_launch(resource, args);
     } else {
-        return resource_managers[resource->rm].launch(resource, args);
+        return_val=resource_managers[resource->rm].launch(resource, args);
     }
+
+    // here we will check the status file that all the servers are
+    // up and talking to each other and are happy.  
+
+    // file stage-in goes here.  
+    // grab the name of the stage-in manifest file.  
+    // check that file for existence, parse it for proper structure
+    // check the files that it refers to
+    // then call the transfer API to transfer all the files
```

Comment:
Note that you can't use the transfer API in the `unifyfs` utility. It may run on a node where the server is not running (for current LSF systems, it runs on a 'batch' node that is not a 'compute' node, which is where the servers are launched). Talk to Hyogi about my suggestion for this.

Reply:
Right, that's a good point.  We'll probably do the parsing locally, read the manifest, check it for validity, and create a set of marching orders corresponding to *each* valid file in the manifest that we want to transfer in.  

For each such file, we'll need to construct a proper jsrun (or appropriate equivalent) command that will fire off the transfer-static (or whatever) binary that, as you say, will then be launched on the *compute* nodes, where the servers live.